src/milia/api/http.cljc

```diff
@@ -29,11 +34,19 @@
                                            raw-response?)]
        (debug-api method url http-options response)
        ;; Assume that a nil status indicates an exception
-       (when (or (nil? status)
-                 (and (>= status 400) (< status 500)
-                      (not suppress-4xx-exceptions?)))
-         (throw+ {:api-response-status status
-                  :parsed-api-response parsed-response}))
+       (cond
+        (nil? response)
+        (throw+ {:reason :no-http-response})
+        (and status
+             (>= status 400)
+             (< status 500)
+             (not suppress-4xx-exceptions?))
+        (throw+ {:reason :http-client-error
```

Comment:
Since the format of what's being thrown is well defined, and we want it to be the same here as on line 47, can we call a helper here and below to generate this map? Like `(format-error-response :http-client-error status parsed-response)`